src/app/content/highlights/components/SummaryPopup/HighlightListElement.tsx

```diff
@@ -81,7 +83,9 @@ interface HighlightListElementProps {
 const HighlightListElement = ({ highlight, locationFilterId, pageId }: HighlightListElementProps) => {
   const [isEditing, setIsEditing] = React.useState(false);
   const [isDeleting, setIsDeleting] = React.useState(false);
+  const book = useSelector(bookSelector);
   const dispatch = useDispatch();
+  const linkToHighlight = React.useMemo(() => createHighlightLink(highlight, book), [highlight, book]);
```

Comment:
can you do this inside `ContextMenu` ?

Reply:
I've moved it from the ContextMenu as Jan suggested to leave it as presentional compoment

Reply:
i suppose "presentational" is a subjective thing, i think it would have been fine to have the link generation in there, but whatever you don't have to put it back